test/abstractarray.jl

```diff
@@ -412,8 +413,8 @@ function test_map(::Type{TestAbstractArray})
     A = Array(Int, 10)
     B = Float64[1:10...]
     C = Float64[1:10...]
-    @test Base.map_to!(f, 1, A, B, C) == Real[ 2 * i for i in 1:10 ]
```

Comment:
This might have been testing the "produce a more concrete output type than `eltype(A)` suggests" behavior.